cgroups/README.md

```diff
@@ -1,73 +1,315 @@
 
-cgroups are specified as flags to `docker run`.  In particular, the following flags are of note for this exercise:
-```
-$ docker run --help
+You will complete the following steps as part of this lab.
+
+- [Step 1 - cgroups and the Docker CLI](#cli)
+- [Step 2 - Max-out two CPUs](#cpu_max)
+- [Step 3 - Set CPU affinity](#cpu_affinity)
+- [Step 4 - CPU share constraints](#cpu-share)
+- [Step 5 - Docker Compose and cgroups](#compose)
+- [Step 6 - Preventing a fork bomb](#fork_bomb)
+
+# Prerequisites
+
+You will need all of the following to complete this lab:
+
+- A Linux-based Docker Host with at least two CPU cores
+- Docker 1.11 or higher
+- The `htop` tool installed on your Docker Host (`apt-get install htop` on Ubuntu hosts)
+- Root access on the Docker Host
+
+
+# <a name="cli"></a>Step 1: cgroups and the Docker CLI
+
+The `docker run` command provides many flags that allow you to apply cgroup limitations to new containers. The following flags are of note for this lab:
+
+   ```
+   $ docker run --help
 ...
   --cpu-shares                    CPU shares (relative weight)
   ...
   --cpuset-cpus                   CPUs in which to allow execution (0-3, 0,1)
   ...
   --pids-limit                    Tune container pids limit (set -1 for unlimited)
-```
+   ```
+
+For more information on cgroup related flags available to the `docker run` command, see the [docker run reference guide](https://docs.docker.com/engine/reference/run/#specifying-custom-cgroups).
+
+# <a name="cpu_max"></a>Step 2: Max-out two CPUs
+
+In this step you'll start a new container that will max out two CPU cores. You will need `htop` installed on your Docker Host, and you will need to be running a Docker Host that has two or more CPU cores. This step will still work if your Docker Host only has a single core. However, some of the `htop` outputs will be slightly different.
+
+1. If you haven't already done so, install `htop` on your Docker Host.
+
+  Ubuntu with `apt`.
+
+  ```
+  $ sudo apt-get install htop
+  ```
+
+  CentOS with `yum`.
+
+  ```
+  $ sudo yum install htop
+  ```
+
+2. Clone the lab's GitHub repo locally on your Docker Host and change into the `cgroups/cpu-stress` directory.
+
+   ```
+   $ sudo git clone https://github.com/riyazdf/dockercon-workshop.git
+   Cloning into 'dockercon-workshop'...
+   remote: Counting objects: 1638, done.
+   Receiving objects: 100% (1638/1638), 8.84 MiB | 2.37 MiB/s, done.
+   Resolving deltas:  22% (33/148)   d 0 (delta 0), pack-reused 1638
+   Resolving deltas: 100% (148/148), done.
+   Checking connectivity... done.
+   Checking out files: 100% (1389/1389), done.
+
+   $ cd dockercon-workshop/cgroups/cpu-stress
+   ```
+
+  > **Note to lab owner. We might want to move this repo to a more official place along with the rest of the HOL guides and then update this doc accordingly.**
```

Comment:
Thanks for the note, I'll check for the best place to put it with other HOL materials 👍